### DISCUSSION TITLE
Self-contained distribution of CSharp.lua

### DISCUSSION THREAD

**Discussion by ghost** [yanghuan/CSharp.lua#495] [Ideas]
Like any .NET application, CSharp.lua could be published as self-contained.

https://learn.microsoft.com/en-us/dotnet/core/deploying/#publish-self-contained

This is more convenient. No need to install a .NET runtime to use it anymore.

**Comment by yanghuan** [comment]
The CSharp.lua compiler will import the .net standard library by default, if not .NET runtime can not found the system dll

**Reply by ghost** [reply]
Publish a .NET application as self-contained is bundle the whole .NET runtime, including the .NET standard library with it.

**Comment by yanghuan** [comment]
https://github.com/yanghuan/CSharp.lua/blob/master/CSharp.lua/Compiler.cs#L82-L92   
The compiler needs to reference the DLL files from the .NET runtime. Currently, the code is using the DLL files from the .NET runtime directory. If you choose to use a self-contained deployment, you will still need to include these DLL files along with your application.

**Reply by ghost** [reply]
Do you mean Microsoft.CodeAnalysis.dll and Microsoft.CodeAnalysis.CSharp.dll? It's fine to include them, though. Compared to the need to install a full .NET runtime, the self-contained deployment is still much more convenient. Btw, isn't self-contained deployment will bundle these DLL with the distribution, too?
